symengine/polynomial.cpp

```diff
@@ -73,30 +69,28 @@ int UnivariateIntPolynomial::compare(const Basic &o) const
     return map_uint_mpz_compare(dict_, s.dict_);
 }
 
-RCP<const Basic> UnivariateIntPolynomial::from_dict(const RCP<const Symbol> &var, map_uint_mpz &&d)
+RCP<const UnivariateIntPolynomial> UnivariateIntPolynomial::from_dict(const RCP<const Symbol> &var, map_uint_mpz &&d)
 {
-    if (d.size() == 1) {
-        if (d.begin()->first == 0)
-            return integer(d.begin()->second);
-        else if (d.begin()->first == 1) {
-            if (d.begin()->second == 0)
-                return zero;
-            else if (d.begin()->second == 1)
-                return var;
-            else
-                return Mul::from_dict(integer(d.begin()->second), {{var, one}});
-        } else {
-            if (d.begin()->second == 0)
-                return zero;
-            else if (d.begin()->second == 1)
-                return pow(var, integer(d.begin()->first));
-            else
-                return Mul::from_dict(integer(d.begin()->second),
-                    {{var, integer(d.begin()->first)}});
+    for(auto itter = d.begin(); itter != d.end(); itter++){
+        if(integer_class(0) == itter->second)
+  	    d.erase(itter);
```

Comment:
This is probably what is causing the segfault. If you delete while iterating, the iterator becomes invalid.